source/ucumLhcUtils.js

```diff
@@ -145,71 +145,157 @@ export class UcumLhcUtils {
   } // end validateUnitString
 
 
+
+/**
+ * @typedef {
+ *   'normal', 
+ *   'mol->mass', 
+ *   'mass->mol', 
+ *   'eq->mass', 
+ *   'mass->eq', 
+ *   'eq->mol', 
+ *   'mol->eq', 
+ * } ConversionType
+ */
+
+  /**
+  * Detects the type of conversion between two units.
+  *
+  * @param {Object} fromUnit - The unit to convert from.
+  * @param {Object} toUnit - The unit to convert to.
+  * @returns {ConversionType} conversionType - The type of conversion as a string.
+  */
+  detectConversionType(fromUnit, toUnit) {
+    /** @type {ConversionType} */
+    let conversionType = 'normal';
+
+    /** Note: 
+     * isMolarUnit() will return true for both molar and equivalent units 
+     * because equivalent units are a subset of molar units. When checking 
+     * for eq<->mol conversions, we need explicitly check that one unit is
+     * an equivalent unit and the other is a molar unit.
+     * 
+     * For this reason, the order of the below checks matters and we need to
+     * check for eq<->mol conversions before mol<->mol conversions.
+     */ 
+
+    // if neither unit is a molar/eq unit, return 'normal'
+    if (!(fromUnit.isMolarUnit() || toUnit.isMolarUnit())) {
```

Comment:
Since isMolarUnit only checks moleExp_, then this is saying that if neither unit has moles, the conversionType is normal, but that is not necessarily true.  It could be eq<->mass.

Reply:
Because of my comment above (eq are a subset of moles), for any eq property both molExp_ and equivalentExp_ will be set. 

Since this is the case, the above isn't a bug as it will catch the case you mention.

Reply:
With that said, I do think theres value to being more explicit in the code - even if the ultimate behavior doesn't change. We could update the line to something similar to the following for clarity:

```js
if (!(fromUnit.isMolarUnit() || toUnit.isMolarUnit()) && !(fromUnit.isEquivalentUnit() || toUnit.isEquivalentUnit())) {
```

Reply:
I didn't see in the code that moleExp_ is set if equivalentExp_ is set, but regardless, I don't think that will be helpful in the next PR when you are allowing those to take on any integer value and doing math with them.  They are different units, even though there is a (variable) conversion factor between them, so I think it is best to keep them separate.

Reply:
> With that said, I do think theres value to being more explicit in the code - even if the ultimate behavior doesn't change. We could update the line to something similar to the following for clarity:
> 
> ```js
> if (!(fromUnit.isMolarUnit() || toUnit.isMolarUnit()) && !(fromUnit.isEquivalentUnit() || toUnit.isEquivalentUnit())) {
> ```
I think what you probably want is to combine a few lines and say something like:
```
if ((fromUnit.isMolarUnit() && toUnit.isMolarUnit()) || (fromUnit.isEquivalentUnit() && toUnit.isEquivalentUnit()))
 conversionType  = 'normal';
else ...
```

Reply:
Currently, moving the fromUnit.isMolarUnit() && toUnit.isMolarUnit() up into the same block as that first (fromUnit.isEquivalentUnit() && toUnit.isEquivalentUnit()) check currently breaks the code due to isEquivalentUnit()   not being exclusive from isMolarUnit(), now classifying some eq conversions as mol conversions.

To fix this, I can change isMolarUnit() to be:
```js
  /**
   * Checks if the given unit is a molar unit.
   * 
   * @returns {boolean} - Returns true if the unit is a molar unit, false otherwise.
   */
  isMolarUnit() {
    return this.moleExp_ !== 0 && !this.isEquivalentUnit();
  } // end isMolarUnit
  ```

Reply:
> I didn't see in the code that moleExp_ is set if equivalentExp_ is set, but regardless, I don't think that will be helpful in the next PR when you are allowing those to take on any integer value and doing math with them. They are different units, even though there is a (variable) conversion factor between them, so I think it is best to keep them separate.

This is specified in the generation of definition files itself. 

## **ucumXmlDocument.js**
```js
    // line below has not changed, 'eq' is has been considered a moleCode previously
    this.moleCodes_ = ['mol', 'eq', 'osm', 'kat', 'U' ];
    ...
    this.equivCodes_ = ['eq'];
```

Therefore any equivalent unit, both moleExp_ and equivalentExp_ will be set during the generation of ucumDefs.json

Looking to whether this should be updated in the def file generation itself or downstream in the code.

Reply:
This is now addressed in my latest commit.

While eq are measured in mols, there would be less confusion if they were treated as exclusive entities instead of eq being a subset of mols. This has been updated in the generation of the def file:

```
    // removed 'eq' 
    this.moleCodes_ = ['mol', 'osm', 'kat', 'U' ];
    ...
    this.equivCodes_ = ['eq'];
```